lib/at_host/at_host.c

```diff
@@ -168,34 +196,54 @@ int at_host_init(enum select_uart uart_id, enum term_modes mode)
 		}
 	}
 
+	k_sem_give(&at_host_start_sem);
+
 	return err;
 }
 
-void at_host_process(void)
+static void process_thread_fn(struct k_sem *sem)
 {
-	static u8_t at_read_buff[UART_RX_BUF_SIZE];
-
-	if (at_rdy) {
-		uart_irq_rx_disable(uart_dev);
-		at_rdy = false;
-		/* Detect at_buff overflow*/
-		if (characters > AT_MAX_CMD_LEN) {
-			LOG_ERR("at_buff overflow\n");
-		} else {
-			at_host_at_socket_write(at_buff, characters);
+	u8_t at_read_buff[UART_RX_BUF_SIZE] = {0};
```

Comment:
Do you plan to have more than one thread spawned for modem processing? I don't see if we need to have this buffer declared on a stack.